src/dx/subscribe/subscription.rs

```diff
@@ -0,0 +1,751 @@
+//! # Subscription module.
+//!
+//! This module contains the [`Subscription`] type, which is used to manage
+//! subscription to the specific entity and attach listeners to process
+//! real-time events triggered for the `entity`.
+
+use spin::RwLock;
+use uuid::Uuid;
+
+use crate::core::{Deserializer, Transport};
+use crate::{
+    core::{DataStream, PubNubEntity},
+    dx::pubnub_client::PubNubClientInstance,
+    lib::{
+        alloc::{
+            string::String,
+            sync::{Arc, Weak},
+            vec,
+            vec::Vec,
+        },
+        collections::HashMap,
+        core::{
+            cmp::PartialEq,
+            fmt::{Debug, Formatter, Result},
+            ops::{Deref, DerefMut},
+        },
+    },
+    subscribe::{
+        event_engine::SubscriptionInput, traits::EventHandler, AppContext, EventDispatcher,
+        EventEmitter, EventSubscriber, File, Message, MessageAction, Presence, SubscribableType,
+        SubscriptionCursor, SubscriptionOptions, SubscriptionSet, Update,
+    },
+};
+
+/// Entity subscription.
+///
+/// # Example
+///
+/// ### Multiplexed subscription
+///
+/// ```rust
+/// use pubnub::{
+///     subscribe::{Subscriber, SubscriptionOptions}, Keyset, PubNubClient, PubNubClientBuilder,
+/// };
+///
+/// # fn main() -> Result<(), pubnub::core::PubNubError> {
+/// let client = // PubNubClient
+/// #     PubNubClientBuilder::with_reqwest_transport()
+/// #         .with_keyset(Keyset {
+/// #              subscribe_key: "demo",
+/// #              publish_key: Some("demo"),
+/// #              secret_key: Some("demo")
+/// #          })
+/// #         .with_user_id("uuid")
+/// #         .build()?;
+/// let channel = client.create_channel("my_channel");
+/// let subscription = channel.subscription(None);
```

Comment:
This line is very confusing. Can you at least add comment what it does?

Reply:
Do you mean this line:
```rust
let subscription = channel.subscription(None);
```
?

Just leave a note that it will create `Subscription` instance for the entity?

Reply:
Yeah exactly.